src/plugins/Library/SpiderIndexUploader.java

```diff
@@ -549,29 +545,28 @@ protected void mergeToFreenet(File diskDir) {
 		String f = this.readStringFrom(new File(diskDir, LAST_DISK_FILENAME));
 		if(f == null) {
             if(diskDir.list().length == 0) {
-                System.err.println("Directory "+diskDir+" is empty. Nothing to merge.");
+                Logger.debug(this, "Directory "+diskDir+" is empty. Nothing to merge.");
                 diskDir.delete();
                 return;
             }
             // Ignore
-            System.err.println("Unable to merge old data "+diskDir);
+            Logger.error(this, "Unable to merge old data "+diskDir);
```

Comment:
If this is really an error condition, shouldn’t the error be logged at the place where the error was first encountered? This does not look like the right place.